tpl/collections/collections.go

```diff
@@ -561,21 +568,95 @@ func (ns *Namespace) Union(l1, l2 interface{}) (interface{}, error) {
 		case reflect.Array, reflect.Slice:
 			r := reflect.MakeSlice(l1v.Type(), 0, 0)
 
-			if l1v.Type() != l2v.Type() {
+			if l1v.Type() != l2v.Type() &&
+				l1v.Type().Elem().Kind() != reflect.Interface &&
+				l2v.Type().Elem().Kind() != reflect.Interface {
 				return r.Interface(), nil
 			}
 
+			var l1vv reflect.Value
 			for i := 0; i < l1v.Len(); i++ {
-				elem := l1v.Index(i)
-				if !ns.In(r.Interface(), elem.Interface()) {
-					r = reflect.Append(r, elem)
+				l1vv = l1v.Index(i)
+				if !ns.In(r.Interface(), l1vv.Interface()) {
+					r = reflect.Append(r, l1vv)
 				}
 			}
 
 			for j := 0; j < l2v.Len(); j++ {
-				elem := l2v.Index(j)
-				if !ns.In(r.Interface(), elem.Interface()) {
-					r = reflect.Append(r, elem)
+				l2vv := l2v.Index(j)
+
+				switch l1vv.Kind() {
+				case reflect.String:
```

Comment:
What I'm doing is this PR is casting to like types.  So, `Union([]int{1, 2}, []interface{}{"3"})` would return `[]int{1, 2, 3}`.  `In` doesn't do that -- it only compares matching types.

Much of this matching logic could be pushed to `In`, but `Union` would still have to get a matching type for the returned slice (whose type matches that of `l1`).

I'm open to suggestions.